src/plugins/chart_expressions/expression_heatmap/public/components/heatmap_component.tsx

```diff
@@ -258,7 +258,7 @@ export const HeatmapComponent: FC<HeatmapRenderProps> = memo(
         const percentageNumber = (Math.abs(value - min) / (max - min)) * 100;
         value = parseInt(percentageNumber.toString(), 10) / 100;
       }
-      return metricFormatter.convert(value);
+      return metricFormatter.convert(value).toString();
```

Comment:
I haven't tested it but I wonder why you are making these changes to the formatters? 🤔

Reply:
If you'll remove all the `toString` (I'll change that to `${value ?? ''}`), and select as valueAccessor, xAccessor and yAccessor some field, which contains numbers, you'll face such an error:

![Screenshot 2021-12-22 at 13 44 07](https://user-images.githubusercontent.com/22456368/147087898-773003a9-9ccb-43d5-8162-b27d9b6eba97.png)

The reason for this error is the way our formatted is working. 

https://github.com/elastic/kibana/blob/7b0ab30de320dc9dd530564a49caa1ec9d35281f/src/plugins/field_formats/common/field_format.ts#L126

If no converter was found, it is returning a value, passed to the arguments (in our case that is number), instead of converting it to string. It is just masking the real type. And later, at the elastic-charts (as the string is expected), `split` is called on the face string (number, in our case).

Reply:
I've encountered this error before in elastic-charts and that time we decided to make the EC responsible to cast the value internally, rather than take care of the problem on the consumer side.